tests/general/MementoBearTest.py

```diff
@@ -0,0 +1,289 @@
+
+    For URLs with no status codes appended, a ``RequestException`` is raised.
+
+    :param request: The ``request`` that the mocker recieves.
+    :return:        A mocked ``Response`` object.
+    """
+
+    # the connection check url needs to be explicitly
+    # set to 200
+    if request.url == MementoBear.check_connection_url:
+        status_code = 200
+    elif request.url.startswith('http://timetravel.mementoweb.org/timegate/'):
+        # All memento timegate urls return 200
+        status_code = 200
+    else:
+        try:
+            status_code = int(request.path_url[-3:])
+        except ValueError:
+            raise requests.exceptions.RequestException('The selected string'
+                                                       'cannot be converted to'
+                                                       'integer')
+
+    resp = requests.Response()
+    resp.raw = io.BytesIO()
+    resp.request = request
+    resp.status_code = status_code
+    return resp
+
+
+def memento_archive_status_mock(m, url, is_archived=True, override_head=True):
+    """
+    Mock the memento result for the given url.
+
+    :param m:             A `request_mock.Mocker()` instance.
+    :param url:           The ``url`` to be mocked.
+    :param is_archived:   True means the url will be marked as
+                          archived.
+    :param override_head: True means the url's head request will be overriden
+                          by this method.
+    """
+    if override_head:
+        m.head(url, headers={})
+    if is_archived:
+        # These two mocked responses will be received by memento_client
+        # and make memento_client returns a fake archived link.
+        # The first response headers will tell memento_client to redirect to
+        # the second one.
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            headers=memento_headers[0],
+            status_code=302)
+        m.head(
+            'http://web.archive.org/web/20170421070937/'
+            'https://www.google.com/',
+            headers=memento_headers[1],
+            status_code=200)
+    else:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            status_code=404)
+
+
+def generate_redirects(m, url, redirect_length=1):
+    """
+    Generate redirects path for the given url.
+
+    :param m:               A `request_mock.Mocker()` instance.
+    :param url:             The ``url`` to be redirected.
+    :param redirect_length: How many redirects will be generated.
+    """
+    m.head(
+        url,
+        headers={'Location': '%s/1' % url},
+        status_code=302)
+
+    if redirect_length > 1:
+        for i in range(redirect_length - 1):
+            m.head(
+                '%s/%s' % (url, str(i + 1)),
+                headers={'Location': '%s/%s' % (url, str(i + 2))},
+                status_code=302
+            )
```

Comment:
this is the only instance of `)` on the next line for a `m.head()` invocation.  Consistency please.